src/canvas-effects/canvas-effect.js

```diff
@@ -1794,60 +1602,98 @@ export default class CanvasEffect extends PIXI.Container {
 		}
 	}
 
+	_createFile() {
+		if (this.data.file === "") {
+			return;
+		}
+
+		let file
+		if (this.data.customRange) {
+			const template = this.template ? [this.template.gridSize, this.template.startPoint, this.template.endPoint] : []
+			file = SequencerFileBase.make(
+				this.data.file,
+				"temporary.range.file",
+				{template},
+			);
+		} else if (Sequencer.Database.entryExists(this.data.file)) {
+			file = Sequencer.Database.getEntry(this.data.file).clone();
+		} else {
+			file = SequencerFileBase.make(this.data.file)
+			this._currentFilePath = this.data.file;
+		}
+
+		if (file.template) {
+			this._template = 
+				foundry.utils.mergeObject(
+					{ gridSize: file.template[0], startPoint: file.template[1], endPoint: file.template[2] },
+					this.data.template
+				)
+			
+		}
+		file.fileIndex = this.data.forcedIndex;
+		file.twister = this._twister;
+		this._file = file
+		this._isRangeFind = file?.rangeFind;
+	}
+
+	_updateCurrentFilePath(showDistanceWarning = false) {
+		if (!this._file) {
+			return;
+		}
+		if (!this.data.stretchTo) {
+			this._currentFilePath = this._file.getFile();
+			return;
+		}
+		let ray = new Ray(this.sourcePosition, this.targetPosition);
+		if (ray.distance === 0 && showDistanceWarning) {
+			lib.custom_error(
+			      "effect",
+			      `stretchTo - You are stretching over a distance of "0", you may be attempting to stretch between two of the same coordinates!`
+			);
+		}
+		this._currentFilePath = this._file.getFileForDistance(ray.distance);
+	}
 
 	/**
 	 * Creates the sprite, and the relevant containers that manage the position and offsets of the overall visual look of the sprite
 	 *
 	 * @private
 	 */
-	_createSprite() {
+	async _createSprite() {
 		this.renderable = false;
 
-		const args = [this.spriteSheet ? this.spriteSheet : PIXI.Texture.EMPTY];
-		if (
-			!this.data.xray &&
-			!this.spriteSheet &&
-			!this.data.screenSpace &&
-			!this.data.screenSpaceAboveUI
-		) {
-			args.push(VisionSamplerShader);
-		}
+		// TODO handle / fix vision sampler shader for advanced vision masking?
+		const useVisionMasking = !this.data.xray && !this.data.screenSpace && !this.data.screenSpaceAboveUI;
+		const shader = useVisionMasking ? VisionSamplerShader : undefined
```

Comment:
Is this variable meant to be unused, based on the TODO?

Reply:
It is unused for now that is correct. Vision masking has somehow be broken for some time now it seems. Best to remove it I guess and just leave the todo?

Reply:
Yeah, leave it for now, I'll talk to devNumbers when I finish my move